components/sup/src/manager/mod.rs

```diff
@@ -692,6 +696,11 @@ impl Manager {
                 self.shutdown(ShutdownReason::Departed);
                 return Err(sup_error!(Error::Departed));
             }
+            if let Some(SignalEvent::Passthrough(Signal::HUP)) = signals::check_for_signal() {
+                outputln!("Supervisor shutting down for signal");
+                self.shutdown(ShutdownReason::Signal);
```

Comment:
This `ShutdownReason` enum and its handling is one of the targets of my current refactoring... it's essentially being passed down ultimately to the individual service-shutdown logic. For this work, it would be good to trace this through and in any match / if/else statements where we have branching logic based on `ShutdownReason`, we should ensure that `ShutdownReason::Signal` behaves like `ShutdownReason::PkgUpdating` (since that's what gets passed through when the Supervisor is shutting down for an update).

My examination of the code thus far suggests that we've likely built up a bit too much complexity around `ShutdownReason`, and I think I'll be simplifying a lot of it (ultimately, I think we'll end up with a 2-member enum that just encapsulates the "should we shutdown the services, too?" question). For the purposes of getting this feature in, though, making the above change here should suffice.

Reply:
I agree wholeheartedly with the complexity around `ShutdownReason` and that's what inspired https://github.com/habitat-sh/habitat/issues/5300 in the first place.

However, I think this is fully plumbed. I followed all the code paths through and I believe it does the correct thing without any additional logic. I could've re-used `ShutdownReason::PkgUpdating`, but that seemed misleading. However, there's nothing that checks for `PkgUpdating`, specifically. Any `ShutdownReason` that *isn't* `LauncherStopping` or `SvcStopCmd` behaves identically.

Do you see the need for a code change somewhere?

Reply:
I was thinking from the perspective of documenting intent more than anything; I suspect you are correct that functionally things are OK as is.

Since I'm going to be working on cleaning this up _now_, I'm not going to forget this context, though, so we can merge now if you like.